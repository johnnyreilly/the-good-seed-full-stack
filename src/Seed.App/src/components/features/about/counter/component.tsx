import * as React from 'react';
import { inject, observer } from 'mobx-react';
import { CounterStore } from '../../../../stores/counterStore';
import { Stores } from '../../../../stores';

export interface ICounterProps {
    counterStore: CounterStore;
}

@inject(
    (stores: Stores) => ({
        counterStore: stores.counterStore
    }) as ICounterProps
)
@observer
export class Counter extends React.Component<Partial<ICounterProps>> {
    handleIncrement = () => {
        this.props.counterStore.increment();
    }

    render() {
        const { count } = this.props.counterStore;
        return (
            <div>
                The count is: {count}

                <button onClick={this.handleIncrement}>Increment counter</button>
            </div>
        );
    }
}
