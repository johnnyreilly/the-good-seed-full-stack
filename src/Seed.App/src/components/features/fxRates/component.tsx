import * as React from 'react';
import { inject, observer } from 'mobx-react';
import * as classNames from 'classnames';
import { FxRatesStore } from '../../../stores/fxRatesStore';
import { Stores } from '../../../stores';
import { Rates } from './rates';
import * as css from '../../../styles/styles.scss';

export interface IFxRatesProps {
    fxRatesStore: FxRatesStore;
}

interface IState {
    date: string;
}

@inject(
    (stores: Stores) => ({
        fxRatesStore: stores.fxRatesStore
    }) as IFxRatesProps
)
@observer
export class FxRatesPage extends React.Component<Partial<IFxRatesProps>, IState> {
    state = {
        date: '2012-06-15'
    } as IState;

    handleDateChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { target: { value: date } } = event;
        this.setState(_prevState => ({ date }));
    }

    handleLoadForDateClick = (_event: React.MouseEvent<HTMLButtonElement>) => {
        this.props.fxRatesStore.loadRatesForDate(this.state.date);
    }

    handleLoadLatestClick = (_event: React.MouseEvent<HTMLButtonElement>) => {
        this.props.fxRatesStore.loadRates();
    }

    render() {
        const { date } = this.state;
        const { rates, ratesForDate, lastLoaded } = this.props.fxRatesStore;
        return (
            <div>
                <div className={classNames(css.field)}>
                    <label className={classNames(css.label)}>Date</label>
                    <div className={classNames(css.control)}>
                        <input className={classNames(css.input)} type="text" value={date} onChange={this.handleDateChange} />
                    </div>
                </div>

                <button
                    className={classNames({
                        [css.button]: true,
                        [css.isSuccess]: !ratesForDate.error,
                        [css.isError]: !!ratesForDate.error,
                        [css.isLoading]: ratesForDate.isRequesting
                    })}
                    onClick={this.handleLoadForDateClick}
                >
                    Load 'em up for given date
                </button>

                <button
                    className={classNames({
                        [css.button]: true,
                        [css.isSuccess]: !rates.error,
                        [css.isError]: !!rates.error,
                        [css.isLoading]: rates.isRequesting
                    })}
                    onClick={this.handleLoadLatestClick}
                >
                    Load latest
                </button>

                <p>Last loaded: {lastLoaded === 'ratesForDate' ? 'Rates for date' : lastLoaded === 'rates' ? 'Latest Rates' : null}</p>

                {lastLoaded === 'ratesForDate' ? <Rates {...ratesForDate} /> : null}
                {lastLoaded === 'rates' ? <Rates {...rates} /> : null}
            </div>
        );
    }
}