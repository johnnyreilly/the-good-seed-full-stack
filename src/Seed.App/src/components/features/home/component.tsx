import * as React from 'react';
import { RouteComponentProps } from 'react-router-dom';

export const HomePage: React.SFC<RouteComponentProps<{}>> = _props => (
    <h3>Welcome home!</h3>
);