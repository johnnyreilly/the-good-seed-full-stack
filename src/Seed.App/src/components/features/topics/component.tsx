import * as React from 'react';
import { RouteComponentProps, Route } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { Topic } from './topic';

export class TopicsPage extends React.Component<RouteComponentProps<{}>> {
    render() {
        const { match } = this.props;
        return (
            <div>
                <h2>Topics</h2>
                <ul>
                    <li>
                        <Link to={`${match.url}/rendering`}>
                            Rendering with React
                        </Link>
                    </li>
                    <li>
                        <Link to={`${match.url}/components`}>
                            Components
                        </Link>
                    </li>
                    <li>
                        <Link to={`${match.url}/props-v-state`}>
                            Props v. State
                        </Link>
                    </li>
                </ul>

                <Route path={`${match.url}/:topicId`} component={Topic} />
                <Route exact={true} path={match.url} render={() => ( <h3>Please select a topic.</h3> )} />
            </div>
        );
    }
}