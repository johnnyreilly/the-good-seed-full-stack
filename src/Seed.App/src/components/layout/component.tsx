import * as React from 'react';
import {
    withRouter,
    Route,
    RouteComponentProps,
    Switch
} from 'react-router-dom';
import { observer, inject } from 'mobx-react';
import DevTools from 'mobx-react-devtools';
import * as classNames from 'classnames';
import { Stores } from '../../stores';
import { SecurityStore } from '../../stores/securityStore';
import { Header } from './header';
import { Menu } from './menu';
import { privateRoutes, publicRoutes } from './routes';
import { NotFound } from '../shared/notFound';
import * as css from '../../styles/styles.scss';
import { config } from '../../config';

interface ILayoutProps extends RouteComponentProps<{}> {
    securityStore: SecurityStore;
}

@withRouter
@inject((stores: Stores) => ({
    securityStore: stores.securityStore
}))
@observer
export class Layout extends React.Component<Partial<ILayoutProps>> {

    renderRoute = ({ path, component, exact }: { path: string; exact?: boolean; component: React.ComponentType<RouteComponentProps<any> | {}> }) => (
        <Route key={path} exact={exact} path={path} component={component} />
    )

    render() {
        const { isLoggedIn } = this.props.securityStore;
        return (
            <div>
                <Header />
                <div className={css.pageBackground}>
                    <Menu />

                    <main className={classNames(css.container, css.section, css.pageContent)} role="main">
                        <Switch>
                            {publicRoutes.map(this.renderRoute)}
                            {isLoggedIn
                                ? privateRoutes.map(this.renderRoute)
                                : null}
                            <Route component={NotFound} />
                        </Switch>
                    </main>
                </div>
                {config.development ? <DevTools /> : null}
            </div>
        );
    }
}
