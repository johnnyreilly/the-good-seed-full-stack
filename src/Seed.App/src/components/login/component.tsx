import * as React from 'react';
import { Redirect, RouteComponentProps } from 'react-router-dom';
import { observer, inject } from 'mobx-react';
import * as classNames from 'classnames';
import { Stores } from '../../stores';
import { SecurityStore } from '../../stores/securityStore';
import FaUser from 'react-icons/lib/fa/user';
import FaLock from 'react-icons/lib/fa/lock';
import * as css from '../../styles/styles.scss';

export interface ILoginPageProps extends RouteComponentProps<{}> {
    securityStore: SecurityStore;
}

interface IState {
    username: string;
    password: string;
    redirectToReferrer: boolean;
}

@inject((stores: Stores) => ({
    securityStore: stores.securityStore
}))
@observer
export class LoginPage extends React.Component<ILoginPageProps, IState> {

    state = {
        username: '',
        password: ''
    } as IState;

    handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { target: { value: username } } = event;
        this.setState(_prevState => ({
            username
        }));
    }

    handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { target: { value: password } } = event;
        this.setState(_prevState => ({
            password
        }));
    }

    handleLogin = async (e: React.MouseEvent<HTMLButtonElement>) => {
        e.preventDefault();
        const { securityStore } = this.props;
        const { username, password } = this.state;
        await this.props.securityStore.login({
            username, password
        });
        if (securityStore.userId) {
            this.setState(_prevState => ({ redirectToReferrer: true }));
        }
    }

    render() {
        const { loginResult } = this.props.securityStore;
        const { from } = this.props.location.state || { from: { pathname: '/' } };
        const { redirectToReferrer } = this.state;

        if (redirectToReferrer) {
            return (
                <Redirect to={from} />
            );
        }

        return (
            <div>
                <p>You must log in to view the page at this location: <code>{from.pathname}</code></p>
                <form>
                    <div className={classNames(css.field)}>
                        <p className={classNames(css.control, css.hasIconsRight)}>
                            <input
                                className={classNames(css.input)}
                                type="text"
                                placeholder="username"
                                value={this.state.username}
                                onChange={this.handleUsernameChange}
                            />
                            <span className={classNames(css.icon, css.isSmall, css.isRight)}>
                                <FaUser />
                            </span>
                        </p>
                    </div>
                    <div className={classNames(css.field)}>
                        <p className={classNames(css.control, css.hasIconsRight)}>
                            <input
                                className={classNames(css.input)}
                                type="password"
                                placeholder="Password"
                                value={this.state.password}
                                onChange={this.handlePasswordChange}
                            />
                            <span className={classNames(css.icon, css.isSmall, css.isRight)}>
                                <FaLock />
                            </span>
                        </p>
                    </div>
                    <div className={css.field}>
                        <p className={css.control}>
                            <button
                                className={classNames({
                                    [css.button]: true,
                                    [css.isSuccess]: !loginResult.error,
                                    [css.isError]: !!loginResult.error,
                                    [css.isLoading]: loginResult.isRequesting
                                })}
                                onClick={this.handleLogin}
                            >
                                Login
                            </button>
                        </p>
                    </div >
                </form >
                {loginResult.error ? <div>Login failed with this message: {loginResult.error}</div> : null}
            </div >
        );
    }
}
