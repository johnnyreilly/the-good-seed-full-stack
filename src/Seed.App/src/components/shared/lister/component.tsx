import * as React from 'react';
import { observer } from 'mobx-react';
import { ListerStore } from '../../../stores/listerStore';

export interface IListerProps {
    listerStore: ListerStore;
}

interface IState {
    text: string;
}

@observer
export class Lister extends React.Component<Partial<IListerProps>, IState> {
    state = {
        text: ''
    };

    handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { target: { value: text } } = event;
        this.setState(_prevState => ({ text }));
    }

    handleAdd = (_event: React.MouseEvent<HTMLButtonElement>) => {
        this.props.listerStore.add(this.state.text);
    }

    render() {
        const list = this.props.listerStore.list;
        return (
            <div>
                <h2>This is a lister</h2>
                <ul>
                    {list.map((entry, index) => <li key={index}>{entry}</li>)}
                </ul>
                <input type="text" value={this.state.text} onChange={this.handleChange} />
                <button onClick={this.handleAdd}>Add this to the list</button>
            </div>
        );
    }
}
