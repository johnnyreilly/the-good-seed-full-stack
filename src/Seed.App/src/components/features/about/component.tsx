import * as React from 'react';
import { inject } from 'mobx-react';
import { IListerProps, Lister } from '../../shared/lister/component';
import { Stores } from '../../../stores';
import { Counter } from './counter/component';

const Lister1 = inject(
    (stores: Stores) => ({
        listerStore: stores.lister1Store
    }) as IListerProps
)(Lister);

const Lister2 = inject(
    (stores: Stores) => ({
        listerStore: stores.lister2Store
    }) as IListerProps
)(Lister);

export class AboutPage extends React.Component {
    render() {
        return (
            <div>
                <h2>About</h2>
                <Counter />
                <h2>Lister 1</h2>
                <Lister1 />
                <h2>Lister 2</h2>
                <Lister2 />
            </div>
        );
    }
}
