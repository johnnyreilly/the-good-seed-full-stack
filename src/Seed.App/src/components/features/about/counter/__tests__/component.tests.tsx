import * as React from 'react';
import { Provider } from 'mobx-react';
import * as renderer from 'react-test-renderer';
import { Counter } from '../component';

function makeStubCounterStore() {
    const stubCounterStore = {
        count: 0,
        increment() {
            this.count += 1;
        }
    };
    return stubCounterStore;
}

describe('counter', () => {
    it('component renders as expected', () => {
        const stubCounterStore = makeStubCounterStore();
        const component = renderer.create(
            <Provider counterStore={stubCounterStore}>
                <Counter />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});