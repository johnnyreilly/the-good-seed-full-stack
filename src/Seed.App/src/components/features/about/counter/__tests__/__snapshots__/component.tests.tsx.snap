// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter component renders as expected 1`] = `
<div>
  The count is: 
  0
  <button
    onClick={[Function]}
  >
    Increment counter
  </button>
</div>
`;
